### PR TITLE
Expose list, table, ResourceLink componens and useK8sModel(s) hooks via plugin SDK

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

File: frontend/public/components/factory/ListPage/ListPageFilter.tsx [hunk 1]
```diff
@@ -40,7 +24,6 @@ const ListPageFilter: React.FC<ListPageFilterProps> = ({
       nameFilterPlaceholder={nameFilterPlaceholder}
       labelFilterPlaceholder={labelFilterPlaceholder}
       onFilterChange={onFilterChange}
-      textFilter={textFilter}
```
Comment (vojtechszocs) [inline frontend/public/components/factory/ListPage/ListPageFilter.tsx:43]: Don't we need this prop passthrough anymore?

**Review by vojtechszocs — CHANGES_REQUESTED** [review]
Also a small nit: in `frontend/public/*` sources, I'd suggest putting `@console/dynamic-plugin-sdk` imports above the relative path imports (seems like the import sort ESLint rule is not applied to `@console/internal` package sources).

File: frontend/public/components/factory/ListPage/ListPageFilter.tsx (hunk 1 above)
Comment (rawagner) [reply frontend/public/components/factory/ListPage/ListPageFilter.tsx:43]: maybe in the future, but in other form.

`textFilter` is a string name of some function in https://github.com/openshift/console/blob/master/frontend/public/components/factory/table-filters.ts#L39

which is very dangerous.

If in the future we will need to provide custom implementation of text filter a consumer will be able to pass a new optional property as a function, not string. The optional property wont break the API

So we are good for now.

**Comment by rawagner** [comment]
/label px-approved
/label docs-approved

There are no user visible changes, and we have approval for dynamic plugin APIs.

**Comment by yapei** [comment]
tested the PR on a cluster created by cluster-bot, accessing `Demo Plguin -> List page` returns pods list page, the filtering works as well as the 'Create Pod' functionality.

Screen Shot 2021-09-02 at 9 13 34 PM

/label qe-approved

**Pull request merged**